### PR TITLE
FM2-401: Support for $lastn-encounters in R3

### PR DESCRIPTION
**Pull request by Medhavi-16**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-401

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Medhavi-16** [comment]
@ibacher @varung-31 PTAL.

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/ObservationFhirResourceProvider.java [hunk 1]
```diff
@@ -159,4 +159,31 @@ public IBundleProvider getLastnObservations(@OperationParam(name = "max") Number
...
+	 * @param max The value of `N`, default value should be one
+	 * @param subjectParam The reference to a patient
+	 * @param patientParam Another way to reference to a patient
+	 * @param code The code(s) to which the observation should belong
+	 * @param category The category to which the observation should belong If neither patient nor
+	 *            subject is specified, then perform search on all patients,
+	 * @return a bundle of observations whose corresponding encounter is among the recent `N` encounters
+	 *         for the specified patient
+	 */
+	@Operation(name = "lastn-encounters", idempotent = true, type = org.hl7.fhir.r4.model.Observation.class)
```
Comment (varung-31) [inline api/src/main/java/org/openmrs/module/fhir2/providers/r3/ObservationFhirResourceProvider.java:176]: We should be using the Observation model for R3.

**Review by varung-31 — CHANGES_REQUESTED** [review]

**Comment by Medhavi-16** [comment]
@varung-31 @ibacher PTAL.

**Review by varung-31 — APPROVED** [review]
LGTM!

**Pull request merged**